### PR TITLE
Added lab 1 a and b

### PR DESCRIPTION
**Pull request by 916-Preda-Andrei**

### CODE REVIEW

File: Lab 1b/Lexic.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+a.Special symbols, representing:
+- operators: + ,- ,* ,/ ,% ,= ,< ,<= ,= ,>=, +=, -=, *=, /=, !, !=
+- separators: [ ], (), :, space 
+- reserved words: else, if, read, while, write, go, to 
+
+b.identifiers:
+-a sequence of letters and digits, such that the first character is a letter; the rule is: 
+<identifier> ::= <letter> | <identifier><letter> | <identifier><digit>
+<letter> ::= a | b | ... | z
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Lexic.txt]: Cant you use upper case letters for the identifiers?

File: Lab 1b/Lexic.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+b.identifiers:
+-a sequence of letters and digits, such that the first character is a letter; the rule is: 
+<identifier> ::= <letter> | <identifier><letter> | <identifier><digit>
+<letter> ::= a | b | ... | z 
+<digit> ::= 0 | 1 |...| 9 
+
+c.constants 
+1.integer - rule: 
+<noconst> ::= +<no> | -<no> | <no> 
+<no> ::= <digit> | <no><digit>
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Lexic.txt]: You do no treat the case where a number starts with 0. For example, in your definition 0123 is a valid number.

File: Lab 1b/Syntax.in [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<program> ::= <cmpdstmt>
+<arraydecl> ::= <identifier> = []
+<cmpdstmt> ::= /* <stmtlist> */
+<stmtlist> ::= <stmt> | <stmt> <stmtlist>
+<stmt> ::= <simplstmt> | <structstmt>
+<simplstmt> ::= <assignstmt> | <iostmt>
+<assignstmt> ::= <identifier> = <expression>
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Syntax.in]: You should add the arraydecl to the assignment definition. Right now you can not initialize an array with this definition

File: Lab 1b/Token.in [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+>=
++=
+-=
+/=
+!
+
+[
+]
+:
+(
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in:17]: Where do you use this in the syntax definition?

File: Lab 1b/Token.in [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+)
+ 
+go
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in:20]: Same as last comment

File: Lab 1b/Token.in [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+to
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in:21]: Same as last comment

File: Lab 1b/Token.in [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+/
+%
+=
+==
+!=
+<
+<=
+>
+>=
++=
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in]: Where do you use this in the syntax definition?

File: Lab 1b/Token.in [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+-=
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in]: Where do you use this in the syntax definition?

File: Lab 1b/Token.in [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+/=
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in]: Where do you use this in the syntax definition?

File: Lab 1b/Token.in [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+!
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in]: Where do you use this in the syntax definition?

File: Lab 1b/Token.in [hunk 11]
```diff
@@ -0,0 +1,30 @@
++
+-
+*
+/
+%
```
Comment (912-1-Cenan-Horea) [inline Lab 1b/Token.in:5]: Where do you use this in the syntax definition?

**Comment by 912-1-Cenan-Horea** [comment]
Please upload your lab1 files as txt instead of pdf so they can be reviewed :)

File: Lab 1b/Token.in (hunk 4 above)
Comment (916-Preda-Andrei) [reply Lab 1b/Token.in:17]: Please check iostmt and factor

**Pull request merged**